File: MWCompiler-example-infboard/src/ch/fhnw/cpib/iml/parser/concsyntaxtree/Term3.java
package ch.fhnw.cpib.iml.parser.concsyntaxtree;

import ch.fhnw.cpib.iml.parser.abssyntaxtree.IAbsSyn.IExpr;
import ch.fhnw.cpib.iml.parser.concsyntaxtree.IConcSyn.IFactor;
import ch.fhnw.cpib.iml.parser.concsyntaxtree.IConcSyn.IRepMultOprFactor;
import ch.fhnw.cpib.iml.parser.concsyntaxtree.IConcSyn.ITerm3;

public class Term3 implements ITerm3 {

  private final IFactor           factor;
  private final IRepMultOprFactor repMultOprFactor;

  public Term3(final IFactor factor, final IRepMultOprFactor repMultOprFactor) {
    this.factor = factor;
    this.repMultOprFactor = repMultOprFactor;
  }

  @Override
  public void appendAsString(final StringBuilder sb, final String indent) {
    this.factor.appendAsString(sb, indent);
    this.repMultOprFactor.appendAsString(sb, indent);
  }

  @Override
  public IExpr toAbsSyn() {
    return this.repMultOprFactor.toAbsSyn(this.factor);
  }

}

File: CompilerMR/IML_Compiler/IMLParser/DataTypes/Concrete/Term3.cs
using IML_Compiler.IMLParser.DataTypes.Abstract;

namespace IML_Compiler.IMLParser.DataTypes.Concrete
{
    class Term3 : ITerm3
    {
        public IFactor Factor { get; private set; }
        public IRepMultOprFactor RepMultOprFactor { get; private set; }

        public Term3(IFactor fac, IRepMultOprFactor opr)
        {
            Factor = fac;
            RepMultOprFactor = opr;
        }

        public IAbsExpression ToAbySyn()
        {
            return RepMultOprFactor.ToAbsSyn(Factor.ToAbySyn());
        }
    }
}
